pkg/instance/instance.go

```diff
@@ -177,6 +179,30 @@ func (i *Instance) makeVFS() error {
 	return err
 }
 
+// AppsFS returns the hidden filesystem associated with the specified
+// application type
+func (i *Instance) AppsFS(appsType apps.AppType) afero.Fs {
+	switch appsType {
+	case apps.Webapp:
+		return i.hiddenFS(vfs.WebappsDirName)
+	case apps.Konnector:
+		return i.hiddenFS(vfs.KonnectorsDirName)
+	}
+	panic(fmt.Errorf("Unknown application type %s", string(appsType)))
+}
+
+func (i *Instance) hiddenFS(dirname string) afero.Fs {
+	fsURL := config.FsURL()
+	switch fsURL.Scheme {
+	case "file", "mem":
+		return afero.NewBasePathFs(afero.NewOsFs(),
+			path.Join(fsURL.Path, i.Domain, dirname))
+	case "swift":
+		panic("Not implemented")
```

Comment:
Not that much, but we have to think it through.

For instance I was wondering if we really wanted to store konnectors and apps in the same way. The konnector be seen as one single application blob/code (could be stored in a tarball) while an app contains multiple assets that can be served independently.

That said, the work should be relatively straightforward.